lib/watir/elements/element.rb

```diff
@@ -767,7 +768,7 @@ def element_call(precondition = nil, &block)
         element_call(:wait_for_exists, &block) if precondition.nil?
         msg = e.message
         msg += '; Maybe look in an iframe?' if @query_scope.iframe.exists?
-        custom_attributes = @locator.nil? ? [] : selector_builder.custom_attributes
+        custom_attributes = !defined?(@locator) || @locator.nil? ? [] : selector_builder.custom_attributes
```

Comment:
When is `!defined?(@locator)` required? Doesn't `@locator.nil?` already sufficiently handle `@locator` not being defined?

Reply:
No, you get a ruby warning that it might not be initialized.